CMakeLists.txt

```diff
@@ -37,12 +37,17 @@ set(SHARED_UTIL_SRC_FOLDER ${CMAKE_CURRENT_LIST_DIR}/src CACHE INTERNAL "this is
 set(SHARED_UTIL_ADAPTER_FOLDER "${CMAKE_CURRENT_LIST_DIR}/adapters" CACHE INTERNAL "this is where the adapters live" FORCE)
 
 include_directories(${SHARED_UTIL_INC_FOLDER})
-
+get_property(dirs DIRECTORY ${CMAKE_CURRENT_SOURCE_DIR} PROPERTY INCLUDE_DIRECTORIES)
+foreach(dir ${dirs})
+  message(STATUS "dir='${dir}'")
+endforeach()
 CHECK_INCLUDE_FILES(stdint.h HAVE_STDINT_H)
 CHECK_INCLUDE_FILES(stdbool.h HAVE_STDBOOL_H)
 
-if ((NOT HAVE_STDINT_H) OR (NOT HAVE_STDBOOL_H))
+if (WINDOWS)
```

Comment:
Do you mean a bad include path? A linker flag shouldn't have any affect on the preprocessor's search for headers.

My issue with this is that you are getting blinkered by the fact that the directory happens to be called windowsce. Regardless of the OS that the build is targeting, if these two files are missing then they need to be provided by the SDK. Your modification will prevent that from happening except on a Windows platform.

By the way, your pull request still contains debug code unless you intended adding the list of include directories to the output. I prefer to use -v passed to gcc for that information.